docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+```
+
+#####<a id="architect-architectureChain"></a>Return - `architectureChain` - type: OBJECT [#](#architect-architectureChain)
+Object to chain architecture methods onto. Defined below.
+
+####<a id="architectureChain"></a>architectureChain - type: OBJECT [#](#architectureChain)
+The `architectureChain` instance returned from the `Architecture.architect` method. It possesses the following methods, each of which return this same architectureChain instance in order to allow method chaining:
+
+#####<a id="architectureChain-itNeeds"></a>Method - `architectureChain.itNeeds` - type: FUNCTION [#](#architectureChain-itNeeds)
+Accepts any number of inputs: stores, output strings, or arrays of stores and/or output strings. Adds a dependency to the `storeToArchitect` for each store mapped to the passed in inputs. Returns this `architectureChain` instance for continued method chaining.
+
+```
+    architectureChain.itNeeds(storeOrInput, additionalStoresOrInputs...);
+```
+
+######<a id="itNeeds-storeOrInput"></a>Parameter - `storeOrInput` - type: STRING, OBJECT, ARRAY of STRINGs and/or OBJECTs - required [#](#itNeeds-storeOrInput)
+Adds a dependency for the store or stores that map to the passed in string. If a store is passed adds a dependency for the passed in store. If an array is passed in adds a dependency for each element in the passed in array. Will do this for each input.
+
+```
+    storeOrInput = 'users'; //OR
+    storeOrInput = userStore; //OR
+    storeOrInput = ['users', roomStore]; //OR
+    storeOrInput = 'users'; additionalStoreOrInput = roomStore;
+
+```
+
+######<a id="itNeeds-architectureChain"></a>Return - `architectureChain` - type: OBJECT [#](#itNeeds-architectureChain)
+Returns this `architectureChain` instance to allow for continued method chaining.
+
+```
+  console.assert(architectureChain === architectureChain.itNeeds('users'));
+```
+
+#####<a id="architectureChain-itOutputs"></a>Method - `architectureChain.itOutputs` - type: FUNCTION [#](#architectureChain-itOutputs)
+Accepts any number of inputs: strings to output, or arrays of strings to output. Maps this `storeToArchitect` to the passed in output(s) so other stores can add dependencies on this store. If another store is already mapped to this output than stores that need the output will become dependent on both of these stores. Returns this `architectureChain` instance for continued method chaining.
+
+```
+    architectureChain.itOutputs(output, additionalOutputs...);
+```
+
+######<a id="itOutputs-output"></a>Parameter - `output` - type: STRING, ARRAY of STRINGs - required [#](#itOutputs-output)
+Registers the store to the passed in string such that other stores that add a need for the string will be dependent on this store. If an array is passed in registers each store in the passed in array. Will do this for each input. Multiple stores can have the same outputs, in which case a store that needs the string will be dependent on every store that outputs it.
```

Comment:
This section "... that add a need ..." in the first sentence reads oddly right now.